### PR TITLE
Refactor code + new open api v3 parser

### PR DESCRIPTION
**Pull request by quangkhoa**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,5 +43,10 @@
     "style-loader": "^0.16.0",
     "webpack": "^2.3.2",
     "webpack-dev-server": "^2.4.2"
+  },
+  "devDependencies": {
```
Comment (brendo) [inline package.json]: I don't think these are `devDependencies`... that said, the whole project's dependencies need fixing 😕

File: src/components/BodyContent/BodyContent.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    };
+  }
+
+  render() {
+    const { description, schema, example } = this.props;
+
+    const { tab } = this.state;
+    return (
+      <div className="bodyContent">
+        <div className="bodyContent-description">{description}</div>
```
Comment (brendo) [inline src/components/BodyContent/BodyContent.js]: It looks like the current CSS leans with kebab-case so let's change the `className` declarations in this component to follow suit.

File: src/parser/open-api-v3/open-api-v3-parser.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  for (const statusCode in responses) {
+    const response = responses[statusCode];
+
+    const uiResponse = {
+      code: statusCode,
+      description: response.description,
+    };
+
+    if (response.content) {
+      const responseContent = response.content['application/vnd.api+json'] || response.content['application/json'];
```
Comment (brendo) [inline src/parser/open-api-v3/open-api-v3-parser.js]: Can you add a TODO here to abstract this logic out?

File: src/parser/open-api-v3/open-api-v3-parser.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+ *
+ * @return {Immutable.Map}
+ */
+function getUIReadyServices(paths, tags) {
+  const services = {};
+
+  tags.forEach((tag) => {
+    let uiMethods = {};
+
+    for (const pathKey in paths) {
```
Comment (brendo) [inline src/parser/open-api-v3/open-api-v3-parser.js]: This, and the following loop are repeated three times in this file. Could it be abstracted to make this easier/more efficient?

File: package.json (hunk 1 above)
Comment (quangkhoa) [reply package.json]: Tidied up `package.json`

File: src/components/BodyContent/BodyContent.js (hunk 2 above)
Comment (quangkhoa) [reply src/components/BodyContent/BodyContent.js]: Good pickup. Updated.

File: src/parser/open-api-v3/open-api-v3-parser.js (hunk 3 above)
Comment (quangkhoa) [reply src/parser/open-api-v3/open-api-v3-parser.js]: Added `TODO` task

File: src/parser/open-api-v3/open-api-v3-parser.js (hunk 4 above)
Comment (quangkhoa) [reply src/parser/open-api-v3/open-api-v3-parser.js]: An initial traversal is required to work out the tags.

I've combined the loops for `navigation` and `services` into one now to improve efficiency.

So there are 2 loops now.

File: src/store/definition/reducer.js [hunk 5]
```diff
@@ -1,11 +1,207 @@
-let initialState = null;
+import { ActionType } from './constants';
+
+import Immutable from 'immutable';
+
+import RequestRecord from '../../records/RequestRecord';
+import PropertyRecord from '../../records/PropertyRecord';
+import ResponseRecord from '../../records/ResponseRecord';
+
+const definition = new Immutable.Map({
```
Comment (brendo) [inline src/store/definition/reducer.js]: What's the purpose in defining this const? It doesn't look like it's used anywhere?

Comment (quangkhoa) [reply src/store/definition/reducer.js]: you're right. It's a thing of the past. Removed.

**Pull request merged**